internal/test/helper.go

```diff
@@ -118,6 +118,20 @@ func SkipIfNotSecure(t *testing.T) {
 	}
 }
 
+// SkipIfPluginMissing skips a test if the required plugin is missing
+func SkipIfPluginMissing(t *testing.T, client *opensearchapi.Client, plugin string) {
```

Comment:
This may be a bit dangerous since we can be having a passing build, but actually no tests for a given plugin running because of some setup problem.

Reply:
The issue is the test against unreleased opensearch where only the core opensearch gets build so not plugin is installed.
The other option would be to check for an ENV or something like that. Maybe a combination of both. If the ENV is set it is okay if the plugins are missing and if the ENV is missing it fails?

Reply:
I think you want to be explicit about it in all cases. For example, run a different test suite or tests in different folders in unreleased version vs. non unreleased version?

Reply:
I am not sure I can follow your idea but it sounds like you want to use the build tags for testing. So we can test opensearchapi only for unreleased opensearch and test opensearch,security,ism for the other things?
This would require us to tag each file.

Reply:
I am not familiar with the mechanisms for this in Go. Tags would work.

In other languages/frameworks we have things like test suites or simply different folders, where you can explicitly say "run all tests" or "run tests in this folder".

Reply:
I added the unreleased option to the make test-integ run to only run tests in the opensearchapi folder.
The opensearchapi commits are issues that came up during the tests.